src/Emulator/Peripherals/Peripherals/MTD/OpenTitan_FlashController.cs

```diff
@@ -0,0 +1,734 @@
+                        "OpenTitan_FlashController/StartOperation: Read");
+                    StartReadOperation();
+                    break;
+                }
+
+                case ControlOp.FlashProgram:
+                {
+                    this.Log(
+                        LogLevel.Noisy,
+                        "OpenTitan_FlashController/StartOperation: Program");
+                    StartProgramOperation();
+                    break;
+                }
+
+                case ControlOp.FlashErase:
+                {
+                    this.Log(
+                        LogLevel.Noisy,
+                        "OpenTitan_FlashController/StartOperation: Erase");
+                    StartEraseOperation();
+                    break;
+                }
+
+                default:
+                {
+                    this.Log(
+                        LogLevel.Warning,
+                        "OpenTitan_FlashController/StartOperation: invalid controlOpValue: 0x{0:X}", operation.Value);
+                    break;
+                }
+            }
+        }
+
+        private void StartReadOperation()
+        {
+            this.Log(
+                LogLevel.Noisy,
+                "OpenTitan_FlashController/StartReadOperation: addrValue = 0x{0:X}",
+                address.Value);
+
+            this.Log(
+                LogLevel.Noisy,
+                "OpenTitan_FlashController/StartReadOperation: reading {0}",
+                flashSelectPartition.Value ? "InfoPartition" : "DataPartition");
+
+            statusRegisterReadEmptyFlag.Value = false;
+            readAddress = address.Value;
+        }
+
+        private void StartProgramOperation()
+        {
+            this.Log(
+                LogLevel.Noisy,
+                "OpenTitan_FlashController/StartProgramOperation: addrValue = 0x{0:X}",
+                address.Value);
+
+            this.Log(
+                LogLevel.Noisy,
+                "OpenTitan_FlashController/StartProgramOperation: programming {0}",
+                flashSelectPartition.Value ? "InfoPartition" : "DataPartition");
+
+            programAddress = address.Value;
+        }
+
+        private void StartEraseOperation()
+        {
+            this.Log(
+                LogLevel.Noisy,
+                "OpenTitan_FlashController/StartEraseOperation: addrValue = 0x{0:X}",
+                address.Value);
+
+            this.Log(
+                LogLevel.Noisy,
+                "OpenTitan_FlashController/StartEraseOperation: eraseing {0}",
+                flashSelectPartition.Value ? "InfoPartition" : "DataPartition");
+
+            uint MpBankCfgValue = RegistersCollection.Read((long)Registers.MpBankCfg);
+            bool flashEraseBank0 = !flashSelectEraseMode.Value && eraseBank0.Value;
+            bool flashEraseBank1 = !flashSelectEraseMode.Value && eraseBank1.Value;
+
+            dynamic eraseFlash = flashSelectPartition.Value ? (dynamic)infoFlash : (dynamic)dataFlash;
```

Comment:
```suggestion
            var eraseFlash = flashSelectPartition.Value ? infoFlash : (IDoubleWordPeripheral)dataFlash;
```

In fact this fragment is repeated in many places - perhaps it could be extracted to a common helper method?